pkg/build/strategies/sti/sti.go

```diff
@@ -317,7 +317,6 @@ func (builder *STI) createBuildEnvironment() []string {
 	env, err := scripts.GetEnvironment(builder.config)
 	if err != nil {
 		glog.V(1).Infof("No user environment provided (%v)", err)
-		return nil
```

Comment:
Yes, it was by mistake and didn't discover it because on my test project I have `.s2i/environment` file.